Firmware/src/modules/tools/temperaturecontrol/TemperatureControl.cpp

```diff
@@ -679,6 +684,11 @@ void TemperatureControl::setPIDd(float d)
 // called in command context
 void TemperatureControl::in_command_ctx(bool)
 {
+    if (this->sensor_type=="max31855")
+    {
+        OutputStream os(1);  //print error messages to uart
```

Comment:
this is NOT allowed.

Reply:
You would use print_to_all_consoles() for an error message.

Reply:
Calling get_raw() is also incorrect, that is not to be used for reading sensors under normal conditions. I understand that may not be clear ;) but as you can see it is only called for M305.